test/Microsoft.AspNet.Mvc.Core.Test/Rendering/HtmlHelperSelectTest.cs

```diff
@@ -858,6 +864,352 @@ public void ListBoxFor_WithUnreleatedExpression_GeneratesExpectedValue()
+            }
+        }
+
+        private class TestHtmlHelper : HtmlHelper
+        {
+            public TestHtmlHelper([NotNull] IModelMetadataProvider metadataProvider)
+                : base(
+                      new Mock<IHtmlGenerator>(MockBehavior.Strict).Object,
+                      new Mock<ICompositeViewEngine>(MockBehavior.Strict).Object,
+                      metadataProvider,
+                      new Mock<IHtmlEncoder>(MockBehavior.Strict).Object,
+                      new Mock<IUrlEncoder>(MockBehavior.Strict).Object,
+                      new Mock<IJavaScriptStringEncoder>(MockBehavior.Strict).Object)
+            {
+            }
+
+            public ModelMetadata Metadata { get; private set; }
+
+            public IEnumerable<SelectListItem> SelectListItems { get; private set; }
+
+            public IEnumerable<SelectListItem> CopiedSelectListItems { get; private set; }
+
+            protected override IEnumerable<SelectListItem> GetSelectList([NotNull] ModelMetadata metadata)
+            {
+                Metadata = metadata;
+                SelectListItems = base.GetSelectList(metadata);
+                if (SelectListItems != null)
+                {
+                    // Perform a deep copy to help confirm the mutable items are not changed.
+                    var copiedSelectListItems = new List<SelectListItem>();
+                    CopiedSelectListItems = copiedSelectListItems;
+                    foreach (var item in SelectListItems)
+                    {
+                        var copy = new SelectListItem
+                        {
+                            Disabled = item.Disabled,
+                            Group = item.Group,
+                            Selected = item.Selected,
+                            Text = item.Text,
+                            Value = item.Value,
+                        };
+
+                        copiedSelectListItems.Add(copy);
+                    }
+                }
+
+                return SelectListItems;
+            }
+        }
+
+        private class ClassWithFields
+        {
+            public const int Zero = 0;
+
+            public const int One = 1;
+        }
+
+        private enum EmptyEnum
+        {
+        }
+
+        private enum EnumWithDisplayNames
+        {
+            [Display(Name = "tres")]
+            Three = 3,
+
+            [Display(Name = "dos")]
+            Two = 2,
+
+            [Display(Name = "uno")]
+            One = 1,
+
+            [Display(Name = "cero")]
+            Zero = 0,
+
+            [Display(Name = "menos uno")]
+            MinusOne = -1,
+
+#if USE_REAL_RESOURCES
+            [Display(Name = nameof(Test.Resources.DisplayAttribute_Name), ResourceType = typeof(Test.Resources))]
+#else
+            [Display(Name = nameof(TestResources.DisplayAttribute_Name), ResourceType = typeof(TestResources))]
+#endif
+            MinusTwo = -2,
```

Comment:
Should we have a case where an enum field has a `[Display(...)]` but doesn't set the name?